src/workerd/api/http.c++

```diff
@@ -2014,6 +2014,42 @@ jsg::Promise<void> Fetcher::delete_(
       js, JSG_THIS, kj::mv(url), kj::mv(subInit), featureFlags));
 }
 
+jsg::Promise<QueueResponse> Fetcher::queue(
+      jsg::Lock& js, kj::String queueName, kj::Array<ServiceBindingQueueMessage> messages) {
```

Comment:
Since `messages` appears to only be used for the for loop, can it be passed as `kj::ArrayPtr<...>` instead here to avoid an unnecessary move? Likewise, can `queueName` be a `kj::StringPtr`?

Reply:
Interesting, I didn't realize there was any extra meaningful cost there. I'd be willing to, but I get some pretty hairy `jsg::TypeWrapper` compilation errors when I try. Should I use a `Sequence` or a `Generator` instead? FWIW [the official `sendBatch` implementation on the queue binding](https://github.com/cloudflare/workerd/blob/e71985d8dde82662e9a790f9972467d8bef0ab7b/src/workerd/api/queue.h#L49) uses a `Sequence`, although for all I know that's suboptimal too.

--

As for `queueName`, we do move it away into the `QueueEvent::Params` below, so I don't think it'd make sense to switch.

Reply:
Ok then leaving them as is is all good.